video/video_samples.rb

```diff
@@ -0,0 +1,236 @@
+        if location.level == :VIDEO_LEVEL
+          puts "Entire video"
+        else
+          segment          = location.segment
+          start_in_seconds = segment.start_time_offset / 1000000.0
+          end_in_seconds   = segment.end_time_offset / 1000000.0
+
+          puts "#{start_in_seconds} through #{end_in_seconds}"
+        end
+      end
+    end
+  end
+
+  puts "Processing video for label annotations:"
+  operation.wait_until_done!
+  # [END analyze_labels_gcs]
+end
+
+def analyze_labels_local path:
+  # [START analyze_labels_local]
+  # path = "Path to a local video file: path/to/file.mp4"
+
+  require "base64"
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client  = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features      = [:LABEL_DETECTION]
+  video_file    = File.read path
+  encoded_video = Base64.encode64 video_file
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video nil, features, input_content: encoded_video do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+
+    annotation_result.label_annotations.each do |label_annotation|
+      puts "Label description: #{label_annotation.description}"
+      puts "Locations:"
+
+      label_annotation.locations.each do |location|
+        if location.level == :VIDEO_LEVEL
+          puts "Entire video"
+        else
+          segment          = location.segment
+          start_in_seconds = segment.start_time_offset / 1000000.0
+          end_in_seconds   = segment.end_time_offset / 1000000.0
+
+          puts "#{start_in_seconds} through #{end_in_seconds}"
+        end
+      end
+    end
+  end
+
+  puts "Processing video for label annotations:"
+  operation.wait_until_done!
+  # [END analyze_labels_local]
+end
+
+def analyze_faces path:
+  # [START analyze_faces]
+  # path = "Path to a video file on Google Cloud Storage: gs://bucket/video.mp4"
+
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features     = [:FACE_DETECTION]
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video path, features do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+
+    annotation_result.face_annotations.each do |face_annotation|
+      segment_counter = 1
+      puts "Thumbnail size: #{face_annotation.thumbnail.length}"
+
+      face_annotation.segments.each do |segment|
+        if segment.start_time_offset == -1 && segment.end_time_offset == -1
```

Comment:
It's a *little bit* more clear without the counter at least?

```ruby
operation = video_client.annotate_video path, features do |operation|
    raise operation.results.message? if operation.error?
    puts "Finished processing."

    # first result is retrieved because a single video was processed
    annotation_result = operation.results.annotation_results.first

    annotation_result.face_annotations.each do |face_annotation|
      puts "Thumbnail size: #{face_annotation.thumbnail.length}"

      face_annotation.segments.each do |segment|
        if segment.start_time_offset == -1 && segment.end_time_offset == -1
          puts "Entire video"
        else
          start_in_seconds = segment.start_time_offset / 1000000.0
          end_in_seconds   = segment.end_time_offset / 1000000.0

          puts "#{start_in_seconds} to #{end_in_seconds}"
        end
      end
    end
  end
```

I can't think of a way to get rid of or clean up this horribly ugliness tho:

```ruby
if segment.start_time_offset == -1 && segment.end_time_offset == -1
```

Take it as is it is for now... 😕